### PR TITLE
Add SDL game controller support

### PR DESCRIPTION
**Pull request by deveee**

### CODE REVIEW

**Comment by MoNTE48** [comment]
@deveee, can you rebase and fix this PR?

**Comment by deveee** [comment]
Rebased and if you mean that android build failed then it needs irrlicht headers from game-controller branch.

File: src/client/game.cpp [hunk 1]
```diff
@@ -2484,11 +2484,17 @@ void Game::updateCameraOrientation(CameraOrientation *cam, float dtime)
 			input->setMousePos(center.X, center.Y);
 	}
 
+#if defined(_IRR_COMPILE_WITH_SDL_DEVICE_)
+	f32 c = m_cache_joystick_frustum_sensitivity * (1.f / 32767.f) * dtime;
+	cam->camera_yaw -= input->sdl_game_controller.getCameraYaw() * c;
+	cam->camera_pitch += input->sdl_game_controller.getCameraPitch() * c;
+#else
 	if (m_cache_enable_joysticks) {
```
Comment (MoNTE48) [inline src/client/game.cpp:2492]: Setting should be respected.

**Comment by MoNTE48** [comment]
Is it safe to just enable joysticks for all devices?

**Comment by MoNTE48** [comment]
@deveee can clang_format issues be fixed?

File: src/gui/modalMenu.cpp [hunk 2]
```diff
@@ -93,6 +97,52 @@ void GUIModalMenu::draw()
...
+#if defined(_IRR_COMPILE_WITH_SDL_DEVICE_)
+#if defined(__ANDROID__) || defined(__IOS__)
+	if (RenderingEngine::get_raw_device()->getCursorControl()->isVisible() &&
+		SDLGameController::isActive())
+		drawCursor();
+#endif
+#endif
+}
+
+void GUIModalMenu::drawCursor()
```
Comment (MoNTE48) [inline src/gui/modalMenu.cpp:110]: `drawCursor()` should be moved inside if defined.

**Pull request merged**